qiskit/transpiler/passes/optimization/_gate_extension.py

```diff
@@ -29,52 +29,56 @@
 
     # FLIP GATES #
     # XGate
-    XGate._postconditions = lambda self, x1, y1: y1 == z3.Not(x1)
-    CXGate._postconditions = lambda self, x1, y1: y1 == z3.Not(x1)
-    CCXGate._postconditions = lambda self, x1, y1: y1 == z3.Not(x1)
+    XGate._postconditions = lambda self, x1, y1: y1 == z3.Not(x1)  # type: ignore[attr-defined]
+    CXGate._postconditions = lambda self, x1, y1: y1 == z3.Not(x1)  # type: ignore[attr-defined]
+    CCXGate._postconditions = lambda self, x1, y1: y1 == z3.Not(x1)  # type: ignore[attr-defined]
```

Comment:
let's remove the `ignore`. The ideas is not to silence mypy, but to move towards a more complete type annotation.

Reply:
I understand the sentiment, but those are obviously on purpose and there is no plan to "fix" them. As such, those are not errors, and IMHO not having them in error list is preferable.